tests/Unit.Tests/Client/HttpClientWrapperTests.cs

```diff
@@ -86,6 +86,24 @@ public void GetAsync_SearchAndCompareApiException()
             act.Should().Throw<SearchAndCompareApiException>().WithMessage(msg);
         }
 
+        [Test]
+        public void GetAsync_SearchAndCompareApi404()
+        {
+            var ub = new UriBuilder("test");
+            var uri = ub.Uri;
+
+            mockMessageHandler
+                .Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage { StatusCode = HttpStatusCode.NotFound });
```

Comment:
question: can we DRY this setup? repeats a lot in the file, maybe extract a method

Reply:
Done